probe-rs/src/debug/registers.rs

```diff
@@ -1,162 +1,243 @@
-use crate::core::Core;
-
-use probe_rs_target::Architecture;
-
-use std::collections::HashMap;
-
-use crate::core::{RegisterDescription, RegisterFile};
+use crate::{
+    core::{Core, RegisterDataType, RegisterFile},
+    Error, RegisterId, RegisterValue,
+};
+
+/// The group name of a register.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum RegisterGroup {
+    /// Core / CPU Registers
```

Comment:
Should this be Platform to match the other usage?

Reply:
I'm open to votes on that. I personally think 'platform' has a broader meaning, and the DWARF spec calls them base registers, so I went with that here, since the debug module is so closely linked with dwarf stuff.

Reply:
Fair enough.  If we're matching terms from the DWARF spec, which makes sense, a comment may help.